examples/resources/hcp_webhook/import.sh

```diff
@@ -0,0 +1,3 @@
+# HCP Webhook can be imported by specifying the webhook resource name
+terraform import hcp_webhook.example webhook/prokect/840e3701-55b6-4f86-8c17-b1fe397303c5/geo/us/webhook/example-webhook
```

Comment:
Is it possible to strip the static values from the required input name and have provider automatically generate the full resource name based off the input? 

For example instead of 
```
terraform import hcp_webhook.example webhook/prokect/840e3701-55b6-4f86-8c17-b1fe397303c5/geo/us/webhook/example-webhook
```

The user could write
```
# Using an explicit project ID, the import ID is:
# {project_id}:{webhook_geo_location}:{webhook_name}
terraform import hcp_webhook.example 840e3701-55b6-4f86-8c17-b1fe397303c5:us:example-webhook
```

Reply:
I like the idea, but I wonder which one could give more work to write. The API provides the resource_name, but `{project_id}:{webhook_geo_location}:{webhook_name}` will always have to be built by the user. 

Would it be helpful if, next to the `import` documentation, the `curl` to get the resource_name with the API is provided?

Reply:
I'm not sure I understand what you mean by "I wonder which one could give more work to write". Are you referring to the effort of the user? 

Generally speaking the input to an import should be something relatively simple for the user to grok from the system under management so the user can quickly construct the required input without prior knowledge about the underlying API for the resources they wish to manage. 

Looking at the channel import it just takes the bucket name and channel name, all of which is accessible via the UI. 

> Would it be helpful if, next to the import documentation, the curl to get the resource_name with the API is provided?

Are you thinking about users who will query the API over users who just want to use Terraform to manage existing webhooks? 

I think for an advance user who is leveraging the API this information would be helpful especially if they want to do some sort of bulk import. But for the average user I don't think that they will access the API for this information. 

As for constructing from the API, by looking at [webhook events page](https://developer.hashicorp.com/hcp/docs/packer/webhooks), it looks like we have an attribute for project_id, resource_name, but not geo location or the simple webhook name. So constructing `{project_id}:{webhook_geo_location}:{webhook_name}` from the API programmatically would take some extra effort. 

I might be over thinking TBH but my first thought was that there are static values in the input that we can probably do away with for the user since that info may not be directly available in the UI. But also use of webhooks is 🧪 so I can't recall what is shown in the UI.

Reply:
I am using `resource_name` because it is a real identifier of the resource, and I am trying to keep the resource aligned with the API as possible. I think `{project_id}:{webhook_geo_location}:{webhook_name}` is a new way to identify the webhook, which differs from the identifier pattern that we have been adopting in the platform and that the webhook service is designed on top. 

https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/resources/service_principal and https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/resources/iam_workload_identity_provider#import are examples of resources that similar to this one it is also using the new identifier and also uses the resource_name to import. 

I agree that there is not enough information on the UI to write the resource_name, which is only because this service adopted the multi-region capability of including geography. I prefer to mitigate that by guiding how to get the resource_name of a webhook instead of working around making this identifier different from the pattern that we are trying to follow.  So I am leaning toward keeping the resource_name and improving the documentation around because of the `geo` part.